### ISSUE TITLE
masked array: The unit of the quantity is stripped.

### ISSUE THREAD

**Issue by jparal** [hgrecco/pint#688]
strange behavior for masked_array. i wonder if this is a bug?

```python
import pint
import numpy as np
units = pint.UnitRegistry(autoconvert_offset_to_baseunit=False)
# OK
np.ma.masked_array(data=[1], mask=[False]) * units.cm
# UnitStrippedWarning: The unit of the quantity is stripped.
np.ma.masked_array(data=[1], mask=[False]) * units('cm')

```

**Comment by dopplershift** [comment]
Duplicate of #633 I think.

**Comment by hgrecco** [comment]
Revisit after #905

**Comment by jthielen** [comment]
This can be closed as a duplication of #633 as @dopplershift mentioned? It wasn't fixed in #905, or the incoming downcast compatibility follow-up, and instead seems to be an upstream issue (https://github.com/numpy/numpy/issues/15200).
